### ISSUE TITLE
When creating a web component, I want to specify a callback that is executed just before my component is updated, so that I can populate data only once for the response after all other operations and updates are done

### ISSUE THREAD

**Issue by pleku** [vaadin/flow#2229]
This is the same as FW8 `Connector::beforeClientResponse`, but instead it should be done as something that is executed only on-demand (instead of always for all) after all user updates have been done. Thus it is similar to `window.requestAnimationFrame(callback)` in JS - only the last one is executed for the next response for that state node.

The API could be: 
```java
StateTree.java
/**
 * @param attachedNode the node that needs to be attached to the state tree to execute callback
 */
public Registrationish beforeClientResponse(StateNode attachedNode, Runnable callback) {...}  
```
There should be a helper method in the UI class which accepts a Component instead.

No need to add end user documentation, will add a separate ticket
